### ISSUE TITLE
An errno Exception?

### ISSUE THREAD

**Issue by CurtTilmes** [Garland-g/Universal-errno#1]
What do you think about including a throwable Raku `Exception` for `errno`?

Not sure if this is along the right path or if there are ways to improve on this interface, but something like this:
```
class X::errno is Exception {
    has $errno = errno;
    method message() { ~$errno }
    method Int() { +$errno }
}
```

Then you could just `die X::errno.new` when you get a Unix-y error.

I suppose I could wrap something like this up into a separate module, but it seems pretty small/simple and could easily be just stuck in here.

**Comment by Garland-g** [comment]
Thank you for the feedback.

I have been thinking about this, but I haven't come up with a way to make this work cleanly yet. I am trying out a few ideas with the goal of adding something like this into this module.

There are two different ways to express errno that I've seen.
1. a function call returns -1 and sets errno.
2. There are some cases where code uses -errno, in cases where setting errno would not work. An example of where this is used is the io_uring interface in the linux kernel.

I would like to support both cases with an Exception class that is similar to this.

**Comment by Garland-g** [comment]
Implemented in 0.1.0.
